erpnext/projects/doctype/task/task.js

```diff
@@ -42,6 +42,13 @@ frappe.ui.form.on("Task", {
 
 	refresh: function (frm) {
 		frm.set_query("parent_task", { "is_group": 1 });
+		frappe.model.with_doctype('Task', () => {
+			const Parent_field = frappe.meta.get_docfield("Task", "status", frm.docname);
+			const field = frappe.meta.get_docfield("Task Project", "status", frm.docname);
+			field.fieldtype = 'Select';
+			field.options = Parent_field.options;
+			frm.fields_dict.filter_fields.grid.refresh();
```

Comment:
- Check this out https://github.com/Bloomstack/frappe/blob/d4ebcf2ca7ca7ffea9023bb7ce310c0595d67c29/frappe/public/js/frappe/form/form.js#L1097

Reply:
set_df_property(fieldname, property, value, docname, table_field) {}

so accordingly i did ("status", "options",  value,  frm.docname, "projects") still not working let me know anything missing this

Reply:
Added select field in status. resolved this issue